Terraform crash - reporting as instructed by the crash

The console output seems to indicate that it crashed on a syntax error while parsing "${var.apply_immediately}"

Attached is the console output and the crash log

[crash.log.txt](https://github.com/hashicorp/terraform/files/147716/crash.log.txt)
[console.txt](https://github.com/hashicorp/terraform/files/147718/console.txt)

Relevant log section:

```
panic: Error reading level config: strconv.ParseBool: parsing "${apply_immediately}": invalid syntax
2016/02/26 05:32:59 [DEBUG] terraform-provider-aws: 
2016/02/26 05:32:59 [DEBUG] terraform-provider-aws: goroutine 134 [running]:
2016/02/26 05:32:59 [DEBUG] terraform-provider-aws: github.com/hashicorp/terraform/helper/schema.(*ResourceData).get(0xc820479f80, 0xc8201c6820, 0x1, 0x1, 0xc8201c6802, 0x0, 0x0, 0x0, 0x0, 0x1490000, ...)
2016/02/26 05:32:59 [DEBUG] terraform-provider-aws: 	/opt/gopath/src/github.com/hashicorp/terraform/helper/schema/resource_data.go:385 +0x20a
2016/02/26 05:32:59 [DEBUG] terraform-provider-aws: github.com/hashicorp/terraform/helper/schema.(*ResourceData).getChange(0xc820479f80, 0x12be470, 0x11, 0x1201, 0x0, 0x0, 0x0, 0x0, 0xc820540000, 0x0, ...)
2016/02/26 05:32:59 [DEBUG] terraform-provider-aws: 	/opt/gopath/src/github.com/hashicorp/terraform/helper/schema/resource_data.go:356 +0x235
2016/02/26 05:32:59 [DEBUG] terraform-provider-aws: github.com/hashicorp/terraform/helper/schema.(*ResourceData).diffChange(0xc820479f80, 0x12be470, 0x11, 0x0, 0x0, 0x0, 0x0, 0x0)
2016/02/26 05:32:59 [DEBUG] terraform-provider-aws: 	/opt/gopath/src/github.com/hashicorp/terraform/helper/schema/resource_data.go:333 +0x78
2016/02/26 05:32:59 [DEBUG] terraform-provider-aws: github.com/hashicorp/terraform/helper/schema.schemaMap.diffString(0xc8205615c0, 0x12be470, 0x11, 0xc8201a95f0, 0xc8203b2e70, 0xc820479f80, 0x0, 0x0, 0x0)
2016/02/26 05:32:59 [DEBUG] terraform-provider-aws: 	/opt/gopath/src/github.com/hashicorp/terraform/helper/schema/schema.go:919 +0xa5
2016/02/26 05:32:59 [DEBUG] terraform-provider-aws: github.com/hashicorp/terraform/helper/schema.schemaMap.diff(0xc8205615c0, 0x12be470, 0x11, 0xc8201a95f0, 0xc8203b2e70, 0xc820479f80, 0x0, 0x0, 0x0)
2016/02/26 05:32:59 [DEBUG] terraform-provider-aws: 	/opt/gopath/src/github.com/hashicorp/terraform/helper/schema/schema.go:566 +0xaf
```

--- a/terraform/interpolate.go
+++ b/terraform/interpolate.go
@@ -284,12 +284,14 @@ func (i *Interpolater) valueSimpleVar(
 	n string,
 	v *config.SimpleVariable,
 	result map[string]ast.Variable) error {
-	// SimpleVars are never handled by Terraform's interpolator
-	result[n] = ast.Variable{
-		Value: config.UnknownVariableValue,
-		Type:  ast.TypeString,
-	}
-	return nil
+	// This error message includes some information for people who
+	// relied on this for their template_file data sources. We should
+	// remove this at some point but there isn't any rush.
+	return fmt.Errorf(
+		"invalid variable syntax: %q. If this is part of inline `template` parameter\n" +
+			"then you must escape the interpolation with two dollar signs. For\n" +
+			"example: ${a} becomes $${a}." +
+			n)
 }
 
 func (i *Interpolater) valueUserVar(

--- a/terraform/interpolate_test.go
+++ b/terraform/interpolate_test.go
@@ -13,6 +13,12 @@ import (
 	"github.com/hashicorp/terraform/config"
 )
 
+func TestInterpolater_simpleVar(t *testing.T) {
+	i := &Interpolater{}
+	scope := &InterpolationScope{}
+	testInterpolateErr(t, i, scope, "simple")
+}
+
 func TestInterpolater_countIndex(t *testing.T) {
 	i := &Interpolater{}
 

